witness/src/Dscp/Snowdrop/Actions.hs

```diff
@@ -5,96 +5,84 @@ module Dscp.Snowdrop.Actions
 import Data.Time.Clock (UTCTime (..))
+import UnliftIO (MonadUnliftIO)
+
 import qualified Snowdrop.Block as SD
+import qualified Snowdrop.Execution as SD (CompositeChgAccum, DbAccessActions (..),
+                                           DbModifyActions (..), SumChangeSet,
+                                           constructCompositeActions)
+import qualified Snowdrop.Util as SD (ExecM, RIO (..))
 
 import Dscp.Core
 import Dscp.Crypto
+import Dscp.DB.Rocks
 import Dscp.Snowdrop.Configuration (BlockPlusAVLComposition, Ids (..), Values (..))
 import Dscp.Snowdrop.Serialise ()
-import Dscp.Snowdrop.Storage.Pure (blockDbActions)
+import Dscp.Snowdrop.Storage.Avlp
+import Dscp.Snowdrop.Storage.Rocks
 import Dscp.Snowdrop.Types
 import Dscp.Witness.AVL (AvlHash (..), AvlProof)
 import Dscp.Witness.Config
-import Snowdrop.Execution (AVLChgAccum, ClientError (..), CompositeChgAccum, DbAccessActions,
-                           RememberForProof, RootHash, avlServerDbActions,
-                           constructCompositeActions, deserialiseM, dmaAccessActions,
-                           initAVLPureStorage)
 
 -- It should be something more complex than IO.
-type SDVars = SDActionsM (RIO LoggingIO)
+type SDVars = SDActionsM SD.ExecM
+
+deriving instance MonadUnliftIO (SD.RIO ctx)
 
 -- Parameter m will be instantiated with RIO Context when the context is defined.
 data SDActionsM m = SDActionsM
-    { nsStateDBActions  :: RememberForProof
-                        -> DbModifyActions (AVLChgAccum AvlHash Ids Values) Ids Values m AvlProof
-    , nsBlockDBActions  :: DbModifyActions (SumChangeSet Ids Values) Ids Values m ()
+    { nsStateDBActions  :: SD.DbModifyActions (AVLChgAccum AvlHash Ids Values) Ids Values m AvlProof
+    , nsBlockDBActions  :: SD.DbModifyActions (SD.SumChangeSet Ids Values) Ids Values m ()
     , nsSDParamsBuilder :: SD.OSParamsBuilder
     }
 
-initSDActions ::
-       forall m n.
-       (MonadIO m, MonadCatch m, MonadIO n, MonadCatch n, HasWitnessConfig)
-    => m (SDActionsM n)
-initSDActions = do
-    avlInitState <- liftIO $ initAVLPureStorage @Ids @Values initAccounts
-    (serverStDba, serverLookupHash) <- avlServerDbActions @Ids @Values @n avlInitState
-    serverBlkDba <- liftIO $ blockDbActions
-    let startTime = UTCTime (toEnum 0) (toEnum 0)
-    let nsSDParamsBuilder = SD.OSParamsBuilder (const $ SD.OSParams startTime startTime)
-    let sdActions = SDActionsM serverStDba serverBlkDba nsSDParamsBuilder
-
-    -- This is something to be used by AVL client (???)
-    let retrieveF :: AvlHash -> m (Maybe ByteString)
-        retrieveF h = serverLookupHash h >>= \resp -> do
-          whenJust resp $ \resp' -> do
-            (respV :: AVL.MapLayer AvlHash Ids Values AvlHash) <- deserialiseM resp'
-            let respProof = AVL.Proof . Free $ pure <$> respV
-            when (not $ AVL.checkProof h respProof) $ throwM BrokenProofError
-          pure resp
-    let (rootHash :: RootHash AvlHash) = gett avlInitState
```

Comment:
Well, you have to remain actions based on pure storage, or how are we going to write tests then?

Reply:
I'll write `Data.Tree.AVL.Store.Pure`-base actions then.

_This should be the same actions as RDB-ones, but ya know, :snowflake: can't into monad transformers_, so I'll have to write another ones.